mesonbuild/modules/cmake.py

```diff
@@ -334,42 +344,34 @@ def create_package_file(self, infile, outfile, PACKAGE_RELATIVE_PATH, extra, con
         shutil.copymode(infile, outfile_tmp)
         mesonlib.replace_if_different(outfile, outfile_tmp)
 
-    @permittedKwargs({'input', 'name', 'install_dir', 'configuration'})
-    def configure_package_config_file(self, state, args, kwargs):
-        if args:
-            raise mesonlib.MesonException('configure_package_config_file takes only keyword arguments.')
-
-        if 'input' not in kwargs:
-            raise mesonlib.MesonException('configure_package_config_file requires "input" keyword.')
+    @noPosargs
+    @typed_kwargs(
+        'cmake.configure_package_config_file',
+        KwargInfo('configuration', build.ConfigurationData, required=True),
+        KwargInfo('input',
+                  (str, mesonlib.File, ContainerTypeInfo(list, (str, mesonlib.File))), required=True,
```

Comment:
```suggestion
                  (str, mesonlib.File, ContainerTypeInfo(list, mesonlib.File)), required=True,
```

I think there's probably no value in letting people pass `input: ['string']` simply because as an internal implementation detail we allow `input: files('string')` and don't force the user to de-arrayize it with `input: files('string')[0]`